tree/treeplayer/src/TBranchProxy.cxx

```diff
@@ -18,13 +18,56 @@ of the autoloading of branches as well as all the generic setup routine.
 #include "TLeaf.h"
 #include "TBranchElement.h"
 #include "TBranchObject.h"
+#include "TCollection.h" // TRangeStaticCast
+#include "TObjArray.h"
 #include "TStreamerElement.h"
 #include "TStreamerInfo.h"
+#include "ROOT/InternalTreeUtils.hxx" // GetFileNamesFromTree, GetTreeFullPaths
+
+#include <algorithm>
+#include <vector>
+#include <string>
 
 ClassImp(ROOT::Detail::TBranchProxy);
 
 using namespace ROOT::Internal;
 
+namespace {
+/**
+ * \brief Get the complete list of branch names from the input branch list.
+ */
+std::vector<std::string> GetBranchNamesRecursive(const TObjArray *branchList)
+{
+   if (!branchList)
+      return {};
+
+   std::vector<std::string> allBranchNames;
+   allBranchNames.reserve(branchList->GetEntries());
+
+   for (auto *b : ROOT::Detail::TRangeStaticCast<TBranch>(branchList)) {
+      allBranchNames.push_back(b->GetName());
+      auto &&subBranchNames = GetBranchNamesRecursive(b->GetListOfBranches());
+      allBranchNames.reserve(allBranchNames.size() + subBranchNames.size());
+      allBranchNames.insert(allBranchNames.end(), std::make_move_iterator(subBranchNames.begin()),
+                            std::make_move_iterator(subBranchNames.end()));
+   }
+
+   return allBranchNames;
+}
+
+/**
+ * \brief Find if the input branch name is the prefix for other sub-branches in the same tree.
+ */
+bool AreThereSubBranches(std::string_view parentBrName, TTree &tree)
+{
+   const auto *branches = tree.GetListOfBranches();
+   const auto allBranchNames = GetBranchNamesRecursive(branches);
+   return std::find_if(allBranchNames.begin(), allBranchNames.end(), [&parentBrName](auto &&brName) {
+             return brName.rfind(parentBrName, 0) != std::string::npos;
```

Comment:
I think it would be safer to test for 'starts with' ("branchname" + "." (if not already there))